integration_tests/src/main/python/cache_test.py

```diff
@@ -285,3 +285,62 @@ def helper(spark):
         return df.selectExpr("a")
 
     assert_gpu_and_cpu_are_equal_collect(helper)
+
+
+@pytest.mark.skipif(is_before_spark_330(), reason='DayTimeInterval is not supported before Spark3.3.0')
+@ignore_order(local=True)
+def test_cache_daytimeinterval_input_row():
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: three_col_df(spark,
+            DayTimeIntervalGen(), int_gen, null_gen).cache().selectExpr('b', 'a'))
+
+
+@pytest.mark.skipif(is_before_spark_330(), reason='DayTimeInterval is not supported before Spark3.3.0')
+@allow_non_gpu("FileSourceScanExec", "ColumnarToRowExec")
+@pytest.mark.parametrize('alongside_gen', [int_gen, ArrayGen(int_gen)], ids=idfn)
+@pytest.mark.parametrize('with_rapids_memoryscan', ['true', 'false'],
+                         ids=["rapids_memoryscan_on", "rapids_memoryscan_off"])
+@pytest.mark.parametrize('with_rapids_reader', ['true', 'false'],
+                         ids=["rapids_reader_on", "rapids_reader_off"])
+def test_cache_daytimeinterval_input_columnar(spark_tmp_path, alongside_gen,
```

Comment:
Why is this test method so complicated? What are we doing in this test that we aren't doing in any other test? We have a DF that we cache and then pull it from the cache, compare the result from the CPU and GPU

In other words, why don't we just do the following? 

```
def test_cache_daytimeinterval_input_columnar():
    def func(spark):
        df = two_col_df(spark, DayTimeIntervalGen(), alongside_gen)
        df.cache().count()
        return df.selectExpr("a") // or whatever the column name is

    assert_gpu_and_cpu_are_equal_collect(func, conf={YOUR CONF})

```

I could be missing something

Reply:
I thought since we implement all the paths (both CPU and GPU) in the PCBS, I think the output for all of the paths should be equal to the original data.
Comparing the GPU output to CPU output is based on the CPU output of the PCBS is reliable and equal to the Spark output.
Maybe it is over designed.

Reply:
Reading data from parquet is to get a columnar input to be cached. 
I do it in the `test_cache_daytimeinterval_input_row` test, it only checks the path of converting internal rows to cached batches.

Reply:
Simplified this  test

Reply:
Reading/writing to parquet isn't necessary to write/read columnar cache. 

If you look at `InMemoryTableScanExec` it calls the `convertColumnarBatchToCachedBatch` if the spark plan supports columnar input and if the serializer supports columnar input (which it always does [here](https://github.com/NVIDIA/spark-rapids/blob/611f54dce73a8e42c575e7ae2f95b1d729085179/sql-plugin/src/main/311%2B-all/scala/com/nvidia/spark/rapids/shims/ParquetCachedBatchSerializer.scala#L266)) 

Reading cache columnar in PCBS depends on three variables, whether the conf `spark.sql.inMemoryColumnarStorage.enableVectorizedReader` is enabled and the plan has 100 or less columns and the plan's output is AtomicType or NullType

Reply:
Good to know.
Just got home and updated the test. Now it does almost the same with `test_cache_columnar`, except the additional data type `DayTimeIntervalType`.